libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/for_each.hpp

```diff
@@ -525,6 +525,20 @@ namespace hpx { namespace ranges {
                 hpx::util::end(rng), std::forward<F>(f),
                 std::forward<Proj>(proj));
         }
+
+        // clang-format off
+        template <typename ExPolicy, typename Predecessor,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::is_execution_policy<ExPolicy>::value)>
+        // clang-format on
+        friend auto tag_fallback_invoke(hpx::ranges::for_each_t,
+            ExPolicy&& policy, Predecessor&& predecessor)
```

Comment:
There probably should be some SFINAE condition for the `Predecessor` as well.

Reply:
Absolutely. My first attempt at requiring it satisfy `is_sender` failed for some (worrying) reason, but given that this is mostly for the sake of the idea at this point I didn't look further into it.

Reply:
> My first attempt at requiring it satisfy is_sender failed for some (worrying) reason

Did you forget to `decay` the argument?

Reply:
Possible... I'll have to give it another shot.